### ISSUE TITLE
autoupdate of Duplicati

### ISSUE THREAD

**Issue by ulicky** [duplicati/duplicati#1673]
It is possible to schedule autoupdate, for example in timeslots, were no backups are running? New versions are so often and I have deployed it on ten machines, because it is so great :)

**Comment by renestach** [comment]
Yes, we should give users an option to automatically install new releases.

**Comment by extremeshok** [comment]
Please could you enable autoupdates.

The 2.0.xx releases would benefit from this.

**Comment by kenkendk** [comment]
Does auto-update "always work" ?

I was under the impression that it occasionally would crash during auto-update, which would be bad.

Should this then work just with an option: "Automatically install new versions when no backups are running" ? Or do we need a full schedule setup?

**Comment by mnaiman** [comment]
It never crashed during update.
Even in case of crash I saw that is already implemented check, that if checksums in update folder are not correct last working version is loaded.
Definitely it has to update when no backups are running, but also there is check that backup is not running during manual update.
On last thing - Duplicati should keep eg 3 version of versions in update folder for easy going back, but not to fill drive with lot of historical versions.

**Comment by agrajaghh** [comment]
I sometimes had problems with the updater in the past, I was in a *loop*.. Duplicati told me there is a new update, I clicked install, the update got downloaded and then I got the notification for the new update again.. I just did a clean install then (no bugreport sry, shame on me)

> On last thing - Duplicati should keep eg 3 version of versions in update folder for easy going back, but not to fill drive with lot of historical versions.

but going back after a database upgrate isnt possible no?

**Comment by mnaiman** [comment]
There is old version of database leftover.
But still better 3 versions than current state of unlimited :)

**Comment by kenkendk** [comment]
Old versions should be deleted automatically:
https://github.com/duplicati/duplicati/blob/master/Duplicati/Library/AutoUpdater/UpdaterManager.cs#L527

It should actually only keep the current and the previous in the update folder?

**Comment by mnaiman** [comment]
When I found out good improvement it is already implemented :)
Ok so we are missing just autoupdate function.

**Comment by mikedunn1** [comment]
Just to confirm, automatic background installation of new versions is still an outstanding enhancement request? The reason I ask is, I just came back to some machines on which I installed Duplicati a year ago and they had not been updated to 2.0.3.3 (although the GUI did prompt me to do so). I wasn't sure if that was a some sort of problem on these machines or expected behavior. My wish list item is silent background updating the way Crashplan does.

Thanks,
Mike

**Comment by mikaelmello** [comment]
+1, will see if I manage do it sometime in the future

**Comment by iBobik** [comment]
> I sometimes had problems with the updater in the past, I was in a loop.. Duplicati told me there is a new update, I clicked install, the update got downloaded and then I got the notification for the new update again.. I just did a clean install then (no bugreport sry, shame on me)

I have this problem on macOS and it happens always.

**Comment by mnaiman** [comment]
Interesting library that could be used for installed and portable version. https://github.com/sn4k3/Updatum
